pkg/documentlog/store/sql.go

```diff
@@ -196,6 +201,25 @@ func (s sqlDocumentStore) GetByConsistencyHash(hash model.Hash) (*model.Document
 	return s.get("consistency_hash = ?", hash.String())
 }
 
+func (s sqlDocumentStore) FindByContentHash(hash model.Hash) ([]model.DocumentDescriptor, error) {
+	s.sqliteMutex.Lock()
```

Comment:
sqlite doesn't have an isolation level?

Reply:
Highly depends on how what you configure in the connection string. By default you can't have parallel access to the database, leading to 2 goroutines having 2 completely different views. If you enable connection sharing you can have concurrent access, but then there's no isolation. So instead of having Nuts Node administrators bug their instance by misconfiguring SQLite I figured we're better off controlling access ourselves.

Until performance becomes an issue I think a mutex does a coarse, but well enough job.

Will have to change when we support other databases like MySQL or PostgreSQL anyways.